### ISSUE TITLE
feature requests?

### ISSUE THREAD

**Issue by naikrovek** [s-u/macosvm#25]
Do you take feature requests?

If I knew Swift, I might contribute these myself, but I don't, yet.

github.com/cirruslabs/tart/ is very much like this but has a much more polished command line interface.  it's not interactive or anything, but it's more mature.  That would be a "nice to have."

One thing I would really, really like, though, is Tart's ability to upload and download VM images from a container registry.  If that is something that's easy to do, I'd like to request that.  If it's not, then never mind, and you can close this issue without response.

I don't mean to make demands of any open source maintainer, so if these feel like things you must do, don't do them, please!  I do not want to be a cause of or contribute to anyone's increasing stress level.  This is a request.  A "no" from you is equally as valid as a "yes."  Like I said, I'm new to Macs and Swift and I intend on taking some of Apple's starter code from WWDC 2022 and expanding it as I learn the language.

Thank you, and have a wonderful [insert time period]!

**Comment by s-u** [comment]
I do take feature requests, just post them as issue here, but they have to provide a bit more details.

What you're talking about above looks like a job for another tool - `macosvm` just runs any images you give it or creates new ones. If you want to upload/download them there are many tools for that - it is not tied to actually running them. However, unlike Docker or similar, it is a bit more tricky, because macOS images are unique (and tied to the configuration, ECID is generated on install) so they are not as easily interchangeable. That said, if anyone wants to donate code that adds interface to some registries, that's fine with me, but as I said, I think that may as well be a separate tool.

**Comment by naikrovek** [comment]
the ECID matters if you're going to run multiple MacOS VMs on the same host, but not if they run on multiple hosts, as I understand it.

given that Apple only allows two simultaneous MacOS VMs on a host, having two separately generated images to run simultaneously was my plan.  

imagine a small (or large) farm of Mac devices running two VMs each as a GitHub Actions farm.  I want to update two container images and know that the next time any of those VM hosts spawns a VM, it will pull the new image first.  that was the scenario in my mind when I created this issue.

Tart's license is no longer Open Source, so I can't take what they've done and use it at the scale I want without sticking to an old version which is AGPL3 or paying for it.  Both of those are difficult sells within an enterprise, so I wanted to put something together that is more acceptable to large businesses.

I've since ceased work on that because of Life, but I'll resume at some point.

**Comment by s-u** [comment]
FWIW you can actually run the same image in parallel in both VMs, it works just fine. The only issues you may run into are Apple-based network services which use the ECID to identify it remotely (e.g. they look as the same VM in sharing), but that doesn't matter for most purposes.

I'll close this since distributing images from a source is trivial and doesn't really need anything from `macosvm` itself, it is fully supported.
